components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -350,29 +362,20 @@ mojom::FeedItemMetadataPtr PickAndRemove(ArticleInfos& articles,
 // selection, but we only consider articles that:
 // 1. The user hasn't subscribed to.
 // 2. **AND** The user hasn't visited.
-mojom::FeedItemMetadataPtr PickDiscoveryArticleAndRemove(
-    ArticleInfos& articles) {
-  PickArticles pick = base::BindRepeating([](const ArticleInfos& articles) {
-    return PickRouletteWithWeighting(
-        articles,
-        base::BindRepeating([](const mojom::FeedItemMetadataPtr& metadata,
-                               const ArticleWeight& weight) {
-          if (weight.subscribed) {
-            return 0.0;
-          }
-          return weight.pop_recency;
-        }));
-  });
-  return PickAndRemove(articles, pick);
+// 3. Don't belong to polarising categories like Politics
+mojom::FeedItemMetadataPtr PickDiscoveryArticleAndRemove(ArticleInfos& articles,
+                                                         PickArticles picker) {
+  return PickAndRemove(articles, std::move(picker));
```

Comment:
Can we move the lambda back into this function? The two usages of it below seem identical.

Alternatively, let's remove this picker function, as it's the same as PickRoulette